spring-grpc-core/src/main/java/org/springframework/grpc/util/GrpcUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.grpc.util;
+
+public class GrpcUtils {
```

Comment:
I think this is `public` mostly because it needs to be used by other packages in this project. 
I would rather not open this up to the world because as you know once its out there...
A freeing move may be to add the word `internal` into the package name which is a signal to the world that it is not a public api (yet). 

The fall-through approach to extract down to the numeric port is brilliant and pragmatic (very important in inception) and it solves our current needs. However, the moment we open it up for external use the edge-casey bugs will come flowing in. Long term I would imagine a regex impl that handles all the variations of bad user input. 

I don't want to figure out that magic regex now. 😄 

Wdyt?

Reply:
We can change the package name.